### DISCUSSION TITLE
DataGrid Editing, how to get edited row values?

### DISCUSSION THREAD

**Discussion by tomironkko** [Megabit/Blazorise#2026] [Q&A] [answered]
Hello,

I am using DataGrid, and I would like to send my Edited row values to Api (PUT). Where can I find those edited values (I click Edit and change the values, and click Save). I would need all the values from the row that I edited.

Thank you very much for your help :)

**Comment by stsrki** [comment, marked as answer]
For this you have the `RowUpdating` and `RowUpdated`, Also `RowInserting` and `RowInserted`.

**Example**

```cs
Task OnRowInserted( SavedRowItem<Employee, Dictionary<string, object>> e )
{
    // ...

    return Task.CompletedTask;
}
```
